ToolChain/Assembler/Parser/test_Parser.py

```diff
@@ -37,978 +35,1368 @@
 
 class TestParser(unittest.TestCase):
 
-    parser = Parser(skipValidation=True)
+    parser = Parser()
+
+    def test_build(self):
+        """
+        Tests the method:
+        build(self, text)
+        """
+
+        self.parser.relativeAddressCounter = 0
+        text = "MOV $B $C"
+        instruction, relAddress, flag = self.parser.build(text)
+        expectedInstruction = bytes((0b10011011,)) + bytes((0b00010010,))
+        expectedRelAddress = 2
+        expectedFlag = 0
+        self.assertEqual(instruction, expectedInstruction)
+        self.assertEqual(relAddress, expectedRelAddress)
+        self.assertEqual(flag, expectedFlag)
+
+        self.parser.relativeAddressCounter = 0
+        text = ".dataAlpha test string  two  spaces\n"
+        instruction, relAddress, flag = self.parser.build(text)
+        expectedInstruction = b'test string  two  spaces\x00'
+        expectedRelAddress = 25
+        expectedFlag = 0
+        self.assertEqual(instruction, expectedInstruction)
+        self.assertEqual(relAddress, expectedRelAddress)
+        self.assertEqual(flag, expectedFlag)
```

Comment:
I think it would read better without the use intermediate values, like so:
```python3
       self.parser.relativeAddressCounter = 0
       text = ".dataAlpha test string  two  spaces\n"
       instruction, relAddress, flag = self.parser.build(text)
       self.assertEqual(instruction, b'test string  two  spaces\x00')
       self.assertEqual(relAddress, 25)
       self.assertEqual(flag, 0)
```

Reply:
This is definitely a better idea, I don't think I was being very efficient here.

Reply:
I don't think you were being inefficient at all! You saved yourself some time using ctrl-c ctrl-v, nothing wrong with that 😆